editor/editor_data.cpp

```diff
@@ -927,9 +951,87 @@ void EditorData::script_class_load_icon_paths() {
 	d.get_key_list(&keys);
 
 	for (List<Variant>::Element *E = keys.front(); E; E = E->next()) {
-		String key = E->get().operator String();
-		_script_class_icon_paths[key] = d[key];
+		String name = E->get().operator String();
+		_script_class_icon_paths[name] = d[name];
+
+		String path = ScriptServer::get_global_class_path(name);
+		script_class_set_name(path, name);
+	}
+}
+
+Ref<Texture> EditorData::get_object_icon(const Object *p_object, const Control *p_control, const String &p_fallback) const {
```

Comment:
This is way too complicated. Make the function part of EditorNode and get the icon from the theme or from the gui_base instance, so you dont have to pass instances around.

Reply:
Done.